packages/react/react/src/hooks/useDebounce.ts

```diff
@@ -12,12 +12,16 @@ import { usePreservedCallback } from './usePreservedCallback';
  *   getV2Logger().log(schemaId, parameter);
  * }, 500);
  */
-export function useDebounce<F extends (...args: any[]) => any>(callback: F, wait: number) {
+export function useDebounce<F extends (...args: any[]) => any>(
+  callback: F,
+  wait: number,
+  options?: Parameters<typeof debounce>[2]
+) {
   const preservedCallback = usePreservedCallback(callback);
 
   const debounced = useMemo(() => {
-    return debounce(preservedCallback, wait);
-  }, [preservedCallback, wait]);
+    return debounce(preservedCallback, wait, options);
+  }, [options, preservedCallback, wait]);
```

Comment:
A reference of `options` parameter can be changed every render because developers may use `useDebounce` like:
```typescript
// passes options object directly
const debounced = useDebounce(callback, { ... });
```

But A debounced callback generated by `lodash/debounce` should keep its own reference. Otherwise debouncing doesn't work as you intended.

So please apply `usePreservedReference` to `options` parameter.
```typescript
import { usePreservedReference } from "./usePreservedReference";

const preservedOptions = usePreservedReference(options);
...
```

Reply:
https://github.com/toss/slash/pull/96

Reply:
Thank you so much for explaining it with an example!
I will take a closer look at the codes for the hooks used and apply them!
Thank you again.🚀

Reply:
You explained it in detail, but I don't think I understood it well.
If you don't mind, I'd like to ask you a question.

```typescript
const [trailing, setTrailing] = useState<boolean>(false);

const debounced = useDebounce(callback, { 
   leading: false;
   trailing: trailing;
});

{...}

setTrailing(true);
```

Are you saying that the second argument of useDebounce is the same reference even though the trailing has changed from false to true value in the above code situation, so debounced callback trailing options is still false? 

If the above situation is correct, I will completely agree with the suggestion and modify the code!

Reply:
No. I have said:
> A reference of options parameter can be changed on every render

Reply:
See the code below.
```typescript
// A
const options = useMemo(() => ({ leading: false, trailing: trailing }), [trailing]);
// B
const options = { leading: false, trailing: trailing };
```
Do you know the difference between `A` and `B`?

Reply:
- `A`
    - `options` object will be newly created only if `trailing` is changed. Even if the component re-rendered, `options` wouldn't be changed without change of `trailing`.
- `B`
    - `options` object will be newly created on every re-render regardless change of its properties.

Reply:
yes I know what useMemo situate

If I don't wrap it with useMemo, the reference of the object can change for each rendering due to a change in the other state, so if the options I sent you didn't actually change, you don't have to receive a new debounced callback

I got it perfectly! I'll fix it!

Reply:
Then you can ask me, "As you say, Developers memoize parameters, don't they?"

But, for developers, it's a good interface to make them pass parameters without considering memoization.

```typescript
useDebonce(setInput, 500, { leading: false }); // good
useDebounce(setInput, 500, useMemo(() => ({ leading: false }), [])); // bad. it's too verbose.
```

So I have suggested you to keep the current interface and apply `usePreservedReference`.

Reply:
As expected, you value the experience of a developer I learned a lot!
and thanks for good examples